# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 16d84439756f
# Full Hash: 16d84439756fae6ab67f7e874dd61d03d3b28905
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r bbfe23c61add -r 16d84439756f dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Mon Jun 08 20:34:15 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Mon Jun 08 20:36:14 2020 +0000
@@ -100,6 +100,7 @@
 
 namespace layers {
 class CompositableHost;
+class SurfaceDescriptor;
 }
 
 namespace webgl {
@@ -484,9 +485,10 @@
   bool PresentInto(gl::SwapChain& swapChain);
 
  public:
-  void Present();
+  void Present(WebGLFramebuffer*, layers::TextureType);
   RefPtr<gfx::DataSourceSurface> GetFrontBufferSnapshot();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebuffer*,
+                                                  layers::TextureType);
 
   void RunContextLossTimer();
   void CheckForContextLoss();